### DISCUSSION TITLE
Google Play Store Publishing - No artifacts were found

### DISCUSSION THREAD

**Discussion by JohannesA1988** [codemagic-ci-cd/codemagic-docs#2001] [Q&A]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

Android (Kotlin)

### Steps to reproduce

Build Android App with this script:
    scripts:
      - name: Build Android release
        working_directory: myapp
        script: |
          LATEST_GOOGLE_PLAY_BUILD_NUMBER=$(google-play get-latest-build-number --package-name "$PACKAGE_NAME")
          if [ -z $LATEST_GOOGLE_PLAY_BUILD_NUMBER ]; then
              # fallback in case no build number was found from google play. Alternatively, you can `exit 1` to fail the build
              UPDATED_BUILD_NUMBER=$BUILD_NUMBER
          else
              UPDATED_BUILD_NUMBER=$(($LATEST_GOOGLE_PLAY_BUILD_NUMBER + 1))
          fi
          ./gradlew bundleRelease -PversionCode=$UPDATED_BUILD_NUMBER -PversionName=1.0.$UPDATED_BUILD_NUMBER
    artifacts:
      - app/build/outputs/**/*.aab
    publishing:
      google_play:
        credentials: $GCLOUD_KEY_FILE
        track: internal
        submit_as_draft: true
        
        
I followed these guides: 
https://docs.codemagic.io/yaml-quick-start/building-a-native-android-app/
https://docs.codemagic.io/yaml-publishing/google-play/

In addition I found that I need to do a first manuell upload. I have done this successfully, including test distribution.

### Expected results

Release should be published to Google Play.

### Actual results

== Gathering artifacts ==
No artifacts were found

### Build id (optional)

64eb8f26a239eb1880e182b6

**Comment by dtrdic** [comment]
hello @JohannesA1988, looks like you are missing some steps before building a release. You first need to Set Android SDK location and install dependencies
```
 - name: Install npm dependencies
    script: |
          npm install
- name: Set Android SDK location
   script: | 
          echo "sdk.dir=$ANDROID_SDK_ROOT" > "$CM_BUILD_DIR/android/local.properties"
```
if you are building a native android app, you can check our [example](https://github.com/codemagic-ci-cd/codemagic-sample-projects/blob/main/react-native/react-native-demo-project/codemagic.yaml#L5) workflow and follow along

**Reply by JohannesA1988** [reply]
Hi @dtrdic, thanks for the fast answer. My app is a native Android App (Kotlin) build with gradle. I checked the corresponding example and here is the build step the same like mine: https://github.com/codemagic-ci-cd/codemagic-sample-projects/blob/main/kotlin-multiplatform-mobile/codemagic.yaml 

But I also tried your approach with npm install and I receive this error:
`
npm ERR! code ENOENT
npm ERR! syscall open
npm ERR! path /Users/builder/clone/package.json
npm ERR! errno -2
npm ERR! enoent ENOENT: no such file or directory, open '/Users/builder/clone/package.json'
npm ERR! enoent This is related to npm not being able to find a file.
npm ERR! enoent 

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/builder/.npm/_logs/2023-08-28T12_19_02_358Z-debug-0.log


Build failed :|
Step 3 script `Install npm dependencies` exited with status code 254`

**Reply by dtrdic** [reply]
oh in that case, try rather with this [example](https://github.com/codemagic-ci-cd/codemagic-sample-projects/blob/main/android/android-native-quick-start/codemagic.yaml).
The install npm dependencies step is not required, but setting Android SDK location it is.

You can check also this step by step [guide](https://docs.codemagic.io/yaml-quick-start/building-a-native-android-app/)

**Comment by dtrdic** [comment]
Hi, I'm tentatively closing the discussion since we haven't heard back from you. You can open the discussion again by replying to this message or opening a new one.
